### PR TITLE
Added an afterSaveAction for SendEmailConfirmation.  Used the SendEma…

### PR DESCRIPTION
**Pull request by enyia21**
…ilConfirmation to send an email after User signs name.

#184899368

### CODE REVIEW

File: src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  Object[] args = new Object[] {submission.getId().toString()};
+  String emailBody = messageSource.getMessage("email.body", args, null);
+  String htmlEmailBody = EmailUtils.wrapHtml(emailBody);
+
+  String recipientEmail = (String) submission.getInputData().get("email");
+  Boolean requireTls = Boolean.TRUE;
+  List<File> pdfs = new ArrayList<File>();
+  try {
+    String generateStringPrefixName = pdfService.generatePdfName(submission.getFlow(), submission.getId().toString());
+    File pdf = File.createTempFile(generateStringPrefixName,".pdf");
```
Comment (bseeger) [inline src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: Is this not the latest code?  It should be using the PDF Service now.  Maybe this is outdated??

Comment (enyia21) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: The pdfService is called on line 47 --> `String generateStringPrefixName = **pdfService**.generatePdfName(submission.getFlow(), submission.getId().toString());`

Comment (bseeger) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: Why not let it generate the PDF itself, too??

Comment (bseeger) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: Oh, I do see that you are calling this below, with creating the FileOutputStream.... hmmm... I need to process this more.

Comment (bseeger) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: Ah, okay, I understand now. I didn't realize you'd still have to create a `File` to put the data into to send to the email service.

Comment (bseeger) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:48]: Makes me wonder if the PdfService should just pass back a `File`. If that would be easier. But don't change things now, we can things later on if it does seem like a File would be better. 

Sorry about all the noise. I hadn't expected you to still need to create a temp file for the PDF  - but this does seem reasonable.

File: src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  private List<String> emailToCc;
+
+  @Value("${form-flow.flow.ubi.email.confirmation.bcc:}")
+  private List<String> emailToBcc;
+public void run(Submission submission){
+
+  String recipientEmail = (String) submission.getInputData().get("email");
+  if (recipientEmail == null || recipientEmail.isBlank()){
+    return;
+  }
```
Comment (bseeger) [inline src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:40]: I think you also need to consider whether or not they want to receive email, too.

Comment (enyia21) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:43]: Got it.  You're right.  I fix it with the latest commit

Comment (bseeger) [reply src/main/java/org/formflowstartertemplate/app/submission/actions/SendEmailConfirmation.java:43]: Great - looks good. Can you please fix the indents in this file?

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -1,4 +1,10 @@
 form-flow:
+  flow:
+    ubi:
+      email:
+        confirmation:
+          cc: testsfdaafwefaewf@gmail.com, testingefafawfe@gmail.com
+          bcc: geowinf@gmil.com
```
Comment (bseeger) [inline src/main/resources/application.yaml:7]: These are good examples, but we should take these lines these out before accepting this ticket, as we don't really have a bcc or cc in UBI. At least I don't think we do and we shouldn't keep random email addresses in here.

If left in, our Demo will spam these hopefully bogus email address and we don't want that.

**Comment by bseeger** [comment]
@enyia21 does this compile locally for you?

**Comment by bseeger** [comment]
This looks like it's good to go once the above comments are addressed.

File: src/main/resources/application.yaml (hunk 3 above)
Comment (enyia21) [reply src/main/resources/application.yaml:7]: This has been updated!

File: src/main/resources/messages.properties [hunk 4]
```diff
@@ -395,3 +395,7 @@ contact-info.select-a-contact-method=In order for us to reach out to you with mo
 generate-pdf-page.header=Generate PDF page!
 generate-pdf-page.title=Generate PDF Page
 generate-pdf-page.button=Generate PDF
+
+email.subject=Your UBI Application
```
Comment (bseeger) [inline src/main/resources/messages.properties:399]: This should be "We received your UBI application" per https://docs.google.com/document/d/1_-2PYwIWRFZsKCZBu8FzmvHhuUzWaAJ40IGKb4EIwwM/edit

**Pull request merged**